observatory_platform/telescopes/crossref_events.py

```diff
@@ -0,0 +1,701 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Author: Aniek Roelofs
+
+import fileinput
+import json
+import logging
+import os
+import pathlib
+import shutil
+from concurrent.futures import ThreadPoolExecutor, as_completed
+from datetime import timedelta
+from multiprocessing import cpu_count
+from typing import Tuple, Union
+
+import pendulum
+from airflow.exceptions import AirflowException
+from airflow.models import Variable
+from airflow.models.taskinstance import TaskInstance
+from google.cloud import bigquery
+from google.cloud.bigquery import SourceFormat
+from requests.exceptions import RetryError
+
+from observatory_platform.utils.config_utils import (AirflowVar,
+                                                     SubFolder,
+                                                     check_variables,
+                                                     find_schema,
+                                                     schema_path,
+                                                     telescope_path,
+                                                     telescope_templates_path)
+from observatory_platform.utils.gc_utils import (bigquery_partitioned_table_id,
+                                                 create_bigquery_dataset,
+                                                 load_bigquery_table,
+                                                 run_bigquery_query,
+                                                 upload_file_to_cloud_storage)
+from observatory_platform.utils.jinja2_utils import (make_sql_jinja2_filename, render_template)
+from observatory_platform.utils.url_utils import retry_session
+
+
+def extract_events(url: str, events_path: str, next_cursor: str = None, success: bool = True) -> \
+        Tuple[bool, Union[str, None], Union[int, None]]:
+    """
+    Extract the events from the given url until no new cursor is returned or a RetryError occurs. The extracted events
+    are appended to a json file, with 1 list per request.
+
+    :param url: The crossref events api url
+    :param events_path: Path to the file in which events are stored
+    :param next_cursor: The next cursor, this is in the response of the api
+    :param success: Whether all events were extracted successfully or an error occurred
+    :return: success, next_cursor and number of total events
+    """
+    headers = {
+        'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_1) AppleWebKit/537.36 (KHTML, like Gecko) '
+                      'Chrome/39.0.2171.95 Safari/537.36'
+    }
```

Comment:
For Crossref Events, it would be more appropriate to set a user agent that identifies our tool, so that we can be contacted and get better service. Having the mailto in the API query might be enough, but we might as well set an appropriate user agent as well. See the page here: https://github.com/CrossRef/rest-api-doc#etiquette

The user agent could be something like: `Observatory Platform (+https://github.com/The-Academic-Observatory/observatory-platform; mailto:agent@observatory.academy)`

Reply:
Or you could set the mailto in the user agent as the same email you use in the mailto in the API calls

Reply:
I've updated the user agent and included the mailto we use in the API calls :-)